rts/lra/client/src/main/java/io/narayana/lra/client/NarayanaLRAClient.java

```diff
@@ -305,7 +305,7 @@ public URI startLRA(URI parentLRA, String clientID, Long timeout, ChronoUnit uni
 
             response = client.target(base)
                 .path(START_PATH)
-                .queryParam(CLIENT_ID_PARAM_NAME, client)
+                .queryParam(CLIENT_ID_PARAM_NAME, URLEncoder.encode(clientID, "UTF-8"))
```

Comment:
Is it wise to encode the clientID? A couple of points:
- it is not necessarily a URL since client ids are arbitrary strings and it may be confusing to the client to receive back something other that what he set on the start call;
- the json data returned by the API call for [returning LRAs](https://github.com/jbosstm/narayana/blob/master/rts/lra/coordinator/src/main/java/io/narayana/lra/coordinator/api/Coordinator.java#L115 ) and the [format of the returned data](https://github.com/jbosstm/narayana/blob/master/rts/lra/service-base/src/main/java/io/narayana/lra/LRAData.java#L42) doesn't say anything about us encoding the client id.

Reply:
Encoding of the `clientID` was my idea. I was afraid that the URL escaped characters won't be correctly transfered and interpreted. I was wrong as I can see from testing the functionality.
Agree that encoding is not necessary, sorry for misguidance @ruromero

Reply:
No worries, I have reverted the change.